### PR TITLE
Discussions | Categories

### PR DESCRIPTION
**Pull request by bkoval**
## Links
* http://wikia-inc.atlassian.net/browse/SOC-1792
* http://wikia-inc.atlassian.net/browse/SOC-1788
* http://wikia-inc.atlassian.net/browse/SOC-1789
* http://wikia-inc.atlassian.net/browse/SOC-1949
* http://wikia-inc.atlassian.net/browse/SOC-1400
etc.

## Description
Categories. Let's do it!

## Reviewers
@Wikia/social-frontend

### CODE REVIEW

File: front/main/app/components/discussion-standalone-editor.js [hunk 1]
```diff
@@ -119,7 +125,13 @@ export default DiscussionMultipleInputsEditor.extend(
 						}
 					}
 
-					this.sendAction(actionName, discussionEntityData);
+					const params = {
+						editedEntity,
```
Comment (rafalkalinski) [inline front/main/app/components/discussion-standalone-editor.js]: Is is still okay, when editedEntity is empty?

File: front/main/app/models/user.js [hunk 2]
```diff
@@ -82,7 +130,71 @@ UserModel.reopenClass({
...
+	 * @param {Ember.Object} model
+	 * @param {QueryUserInfoResponse} query
+	 * @returns {void}
+	 */
+	setUserRights(model, {query}) {
+		const rights = {},
+			rightsArray = query.userinfo.rights;
+
+		if (Ember.isArray(rightsArray)) {
+			// TODO - we could use contains instead of making an object out of an array
```
Comment (rafalkalinski) [inline front/main/app/models/user.js]: what about this?

File: front/main/app/styles/component/_discussion-category-picker.scss [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+		display: inline-block;
+		font-size: 12px;
+		line-height: 12px;
+		margin: 6px 3px;
+		min-width: 54px;
+		padding: 11px 12px;
+		text-align: center;
+
+		&:hover {
+			color: #fff;
```
Comment (rafalkalinski) [inline front/main/app/styles/component/_discussion-category-picker.scss]: color to _settings.scss?

File: front/main/app/templates/components/discussion-categories-edit.hbs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+						<li>
+							<span class="discussion-categories-input-wrapper">
+								{{#if (equal localCategory.id wikiId)}}
+									{{svg 'lock' viewBox='0 0 18 18' class='icon lock'}}
+								{{else}}
+									{{wikia-ui-components/fancy-check-box
+										disabled=true
+									}}
+								{{/if}}
+								{{!TODO use flag instead of equal when backend starts providing it}}
```
Comment (rafalkalinski) [inline front/main/app/templates/components/discussion-categories-edit.hbs]: what about this?

**Comment by rafalkalinski** [comment]
after addressing the minor comments: LGTM!
:+1: :rainbow: 
kalina approved :umbrella:

File: front/main/app/models/user.js (hunk 2 above)
Comment (bkoval) [inline front/main/app/models/user.js]: Not our code. Just moved. Any complaints go to X-Wing :)

File: front/main/app/components/discussion-standalone-editor.js (hunk 1 above)
Comment (bkoval) [inline front/main/app/components/discussion-standalone-editor.js]: Yeah, it's fine. We will not use it when creating a post / reply which would be the case with the empty editEntity.

**Pull request closed without merging**